src/Features/CSharp/Portable/MakeStructFieldsWritable/CSharpMakeStructFieldsWritableDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,102 @@
+﻿
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeQuality;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeStructFieldsWritable
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class CSharpMakeStructFieldsWritableDiagnosticAnalyzer : AbstractCodeQualityDiagnosticAnalyzer
+    {
+        private static readonly DiagnosticDescriptor s_diagnosticDescriptor = CreateDescriptor(
+            IDEDiagnosticIds.MakeStructFieldsWritable,
+            new LocalizableResourceString(nameof(FeaturesResources.Make_readonly_fields_writable), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
+            new LocalizableResourceString(nameof(FeaturesResources.Make_readonly_fields_writable), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
+            isUnneccessary: false);
+
+        public CSharpMakeStructFieldsWritableDiagnosticAnalyzer()
+            : base(ImmutableArray.Create(s_diagnosticDescriptor), GeneratedCodeAnalysisFlags.ReportDiagnostics)
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+        {
+            return DiagnosticAnalyzerCategory.SemanticDocumentAnalysis;
+        }
+
+        public override bool OpenFileOnly(Workspace workspace) => true;
+
+        protected override void InitializeWorker(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(compilationStartContext
+                => SymbolAnalyzer.CreateAndRegisterActions(compilationStartContext));
+        }
+
+        private sealed class SymbolAnalyzer
+        {
+            private bool _hasTypeInstanceAssigment = false;
```

Comment:
no need for false.

Reply:
Resolved in [https://github.com/dotnet/roslyn/pull/32063/commits/5f00e097f529b82deebe0783a03448f9c954f6ec](https://github.com/dotnet/roslyn/pull/32063/commits/5f00e097f529b82deebe0783a03448f9c954f6ec)